TopMovies/API/MovieTarget.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  MovieTarget.swift
+//  TopMovies
+//
+//  Created by anikolaenko on 04.02.2021.
+//
+import ReSwift
+import Moya
+
+class MovieTarget{
+  @Inject private var mainStore: MainStore
+  private let requestedTarget: MoviesDBTarget
+  private var key: String {
+    mainStore.state.configurationState.configuredAPIKey ?? ""
+  }
+  
+  init(requestedTarget: MoviesDBTarget) {
+    self.requestedTarget = requestedTarget
+  }
+  
+  enum MoviesDBTarget {
+    case category(id: MovieCategory.ID, page: Int)
+    case movie(id: MoviePreview.ID)
+    
+    func requestedCategoryPage() -> Int? {
+      guard case let .category(_, page) = self else { return nil }
+      return page
+    }
+  }
+}
```

Comment:
Can we avoid that class? We won't store MovieTarget anywhere, we should only pass it as value type with details for URLRequest (url, path etc.).